gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaAvroJobStatusMonitor.java

```diff
@@ -165,6 +177,26 @@ public GobblinTrackingEvent deserializeEvent(DecodeableKafkaRecord<byte[],byte[]
         break;
       case TimingEvent.JOB_COMPLETION_PERCENTAGE:
         properties.put(TimingEvent.JOB_LAST_PROGRESS_EVENT_TIME, properties.getProperty(TimingEvent.METADATA_END_TIME));
+        break;
+      case JobEvent.WORK_UNITS_CREATED:
+        Long numWorkUnits = Long.parseLong(properties.getProperty(CountEventBuilder.COUNT_KEY));
+        String workUnitCountName = MetricRegistry.name(ServiceMetricNames.GOBBLIN_SERVICE_PREFIX,
+            properties.getProperty(TimingEvent.FlowEventConstants.FLOW_GROUP_FIELD),
+            properties.getProperty(TimingEvent.FlowEventConstants.FLOW_NAME_FIELD),
+            JobEvent.WORK_UNITS_CREATED);
+
+        SortedMap<String, Gauge> existingGauges = this.getMetricContext().getGauges(
+            (name, metric) -> name.equals(workUnitCountName));
```

Comment:
We are doing a full scan of metrics list here, so if there are 100k of them, it can have a performance impact. You can run a profiler on a large deployment to see if a lot of time is spent here. If it is, then we can add a method to check metric existence by name without scanning. Don't see such method in the current library.